pkg/scheduler/scheduler.go

```diff
@@ -770,13 +723,71 @@ func (s *Scheduler) rescheduleResourceBinding(resourceBinding *workv1alpha1.Reso
 	}
 	klog.Infof("The final binding.Spec.Cluster values are: %v\n", resourceBinding.Spec.Clusters)
 
-	_, err = s.KarmadaClient.WorkV1alpha1().ResourceBindings(ns).Update(context.TODO(), resourceBinding, metav1.UpdateOptions{})
+	_, err = s.KarmadaClient.WorkV1alpha1().ResourceBindings(resourceBinding.Namespace).Update(context.TODO(), resourceBinding, metav1.UpdateOptions{})
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func calcReservedCluster(total, ready sets.String) sets.String {
+	return total.Difference(total.Difference(ready))
+}
+
+func calcAvailableCluster(total, ready sets.String) sets.String {
+	return ready.Difference(total)
+}
```

Comment:
I can't see the `total` and `ready` meaning if I just look at the function. It's a math thing.

Consider rename it or add some comments.

Reply:
modify like this:

```go
// bindClusters: set of cluster which exist in resourceBinding/clusterResourceBinding's spec.
// readyClusters: set of cluster whose health status is ready.
// calcReservedCluster calculate set of cluster whose health status is ready in bindClusters.
func calcReservedCluster(bindClusters, readyClusters sets.String) sets.String {
	return bindClusters.Difference(bindClusters.Difference(readyClusters))
}

// bindClusters: set of cluster which exist in resourceBinding/clusterResourceBinding's spec.
// readyClusters: set of cluster whose health status is ready.
// calcAvailableCluster calculate set of cluster whose health status is ready and not in bindClusters.
func calcAvailableCluster(bindCluster, readyClusters sets.String) sets.String {
	return readyClusters.Difference(bindCluster)
}
```